pilot/pkg/bootstrap/server.go

```diff
@@ -1224,7 +1226,51 @@ func (s *Server) initMeshHandlers() {
 	})
 }
 
-func (s *Server) initWorkloadTrustBundle() {
+func (s *Server) addIstioCAToTrustBundle(args *PilotArgs) {
+	// TODO: Will need to watch secrets dynamically
+	// mirrors createIstioCA
+
+	signingKeyFile := path.Join(LocalCertDir.Get(), "ca-key.pem")
+	if _, err := os.Stat(signingKeyFile); err != nil && s.kubeClient != nil {
+		// Fetch self signed certificates
+		err := retry.UntilSuccess(func() error {
+			caSecret, scrtErr := s.kubeClient.CoreV1().Secrets(args.Namespace).
```

Comment:
I may not be understanding the tradeoffs here, but it seems worse to have duplicated (and not identical logic) for the same code. The logic in the CA cod is already super complex, is there any reason we should not update it there as well? is it just time?

Reply:
Agreed that there is a lot of code duplication - added mainly for time constraints. The eventual goal for this is 
(a) move the ENABLE_CA_SERVER flag such that CA setup happens ALWAYS (i.e reads from the . ENABLE_CA_SERVER flag only controls whether the CA server is up and running. This is not a straightforward change, and could take some time. 
(b) Once (a) is done, revert back the older logic of just getting the root certificate from CA.
(c) Have the CA dynamically change its signing cert based on the content of the caCerts secret. The same logic for the trustbundle library to disseminate the root-cert when caCert content changes